crates/sui-adapter/src/adapter.rs

```diff
@@ -921,24 +935,37 @@ fn is_primitive_type_tag(t: &TypeTag) -> bool {
     match t {
         TypeTag::Bool | TypeTag::U8 | TypeTag::U64 | TypeTag::U128 | TypeTag::Address => true,
         TypeTag::Vector(inner) => is_primitive_type_tag(inner),
-        TypeTag::Signer | TypeTag::Struct(_) => false,
+        TypeTag::Struct(StructTag {
+            address,
+            module,
+            name,
+            type_params: type_args,
+        }) => {
+            let resolved_struct = (address, module.as_ident_str(), name.as_ident_str());
+            // is id or..
```

Comment:
what is this comment about?

Reply:
is id or... is option of a primitive (down below)